util/secret/certificates.go

```diff
@@ -456,3 +521,32 @@ func newSelfSignedCACert(key *rsa.PrivateKey) (*x509.Certificate, error) {
 	c, err := x509.ParseCertificate(b)
 	return c, errors.WithStack(err)
 }
+
+func newClientCert(cn string, caCert *x509.Certificate, key *rsa.PrivateKey, caKey *rsa.PrivateKey) (*x509.Certificate, error) {
+	cfg := certs.Config{
+		CommonName: cn,
+	}
+
+	now := time.Now().UTC()
+
+	tmpl := x509.Certificate{
+		SerialNumber: new(big.Int).SetInt64(0),
```

Comment:
This may be pedantic...

>  The serial number MUST be a positive integer assigned by the CA to
>  each certificate.  It MUST be unique for each certificate issued by a
>  given CA (i.e., the issuer name and serial number identify a unique
>  certificate).  CAs MUST force the serialNumber to be a non-negative
>  integer.

-- https://tools.ietf.org/html/rfc3280#section-4.1.2.2

If we fix this, we could also fix it here: https://github.com/kubernetes-sigs/cluster-api/blob/f248367309a112b021c95bd8747b17200739c18a/util/secret/certificates.go#L437

Reply:
It looked wrong to me, but I didn't think to question it further.